StripePayments/StripePayments/Source/PaymentHandler/STPPaymentHandler.swift

```diff
@@ -1311,6 +1311,14 @@ public class STPPaymentHandler: NSObject {
             defer {
                 blockingDataTaskSemaphore.signal()
             }
+
+            // Check if the request failed due to attempting to redirect to an app universal link for Cash App
```

Comment:
@wooj-stripe this is a bit hacky.. when we do follow redirects we get an error back saying the cash app universal link is invalid, so we extract it out of the `error`. We never make a direct request to the universal link via a URL session so this seems like the best way to read it out. thoughts?

The link looks like: `cashme://cash.app/f/...`

Reply:
This is odd, this means that the URL that is vended to us in `authenticationAction.cashAppRedirectToApp?.mobileAuthURL` resulting in an error if we attempt to fetch it?  Why are then vending us a URL which will result in an error?

Reply:
I think this is because URLSession automatically follows redirects and eventually errors when it gets to the universal link. The URL we get back is `https://pm-redirects.stripe.com/authorize...` which redirects to the cash app universal link.

Reply:
`followRedirects` should make a call to the value of  `authenticationAction.cashAppRedirectToApp?.mobileAuthURL`, which should be: `https://pm-redirects.stripe.com/authorize...`, and because that link is expected to be a redirect, the responseURL should be the cash app URL, however, it sounds doing an HTTP GET on `https://pm-redirects.stripe.com/authorize...` is resulting in an error.

1.  Is it erroring because cash app isn't using universal links? (e.g. their link is `cashme://` which is not a universal link.  `cashme://` is a "custom URL Scheme").  Is there any additional info on the error we get back?
  
2.  I thought cash app was using universal links (https://cash.app), which is far superior with regards to security (and provides a much better fall back experience in the event the app isn't installed).  Is there any reason they moved away from universal links?

Reply:
@sfriedman-stripe I asked yesterday but haven't heard back. We only became aware of this within the last 24 hours when we started sending the IN_APP channel. Before that we didn't have insight into the links they were using.